pkg/codegen/go/gen.go

```diff
@@ -1638,6 +1638,13 @@ func (pkg *pkgContext) genResource(w io.Writer, r *schema.Resource, generateReso
 		fmt.Fprint(w, "\topts = append(opts, replaceOnChanges)\n")
 	}
 
+	// Setup PluginDownloadURL. We pre-append our call to PluginDownloadURL to
+	// that if a user adds their own separate PluginDownloadURL, we won't
+	// override it.
+	if url := pkg.pkg.PluginDownloadURL; url != "" {
+		fmt.Fprintf(w, "\topts = append([]pulumi.ResourceOption{pulumi.PluginDownloadURL(%q)},  opts...)\n", url)
```

Comment:
I wonder if we should emit a utility function in the generated `pulumiUtilities.go` for this (only if `PluginDownloadURL` is set), to reduce the amount of duplicated code for each resource.

Reply:
We'll also need to do something similar for invokes.

Reply:
I'll add the function to `pulumiUtilities`. It will make the code marginally smaller. It will also provide a convent place to put the default version if we want that.